src/datasources/cache/cache.router.ts

```diff
@@ -554,4 +564,47 @@ export class CacheRouter {
       `${args.method}_${args.params}`,
     );
   }
+
+  static getStakingDeploymentsCacheDir(): CacheDir {
+    return new CacheDir(this.STAKING_DEPLOYMENTS_KEY, '');
+  }
+
+  static getStakingNetworkStatsCacheDir(): CacheDir {
+    return new CacheDir(this.STAKING_NETWORK_STATS_KEY, '');
+  }
+
+  static getStakingDedicatedStakingStatsCacheDir(): CacheDir {
+    return new CacheDir(this.STAKING_DEDICATED_STAKING_STATS_KEY, '');
+  }
+
+  static getStakingPooledStakingStatsCacheDir(pool: `0x${string}`): CacheDir {
+    return new CacheDir(`${this.STAKING_POOLED_STAKING_STATS_KEY}_${pool}`, '');
+  }
+
+  static getStakingDefiVaultStatsCacheDir(args: {
+    chainId: string;
+    vault: `0x${string}`;
+  }): CacheDir {
+    return new CacheDir(
+      `${args.chainId}_${this.STAKING_DEFI_VAULT_STATS_KEY}_${args.vault}`,
+      '',
+    );
+  }
+
+  /**
+   * Calculate cache directory for staking stakes.
+   *
+   * Note: This function hashes the validators public keys to keep the
+   * cache key short and deterministic.
```

Comment:
Nit: I suggest we add a _why_ here for future reference.

Reply:
Sure, good point! I've tried to clarify it in [dfefc67](https://github.com/safe-global/safe-client-gateway/pull/1897/commits/dfefc678f5daea42ebc3cf85a592c5f1acd90db1)

Let me know if you think it is not enough or if it's not clear.